### PR TITLE
Create port on pin

### PR DESCRIPTION
**Pull request by svandenb-dev**
Added feature to create port on given pins and reference pins on component.

### CODE REVIEW

**Comment by ring630** [comment]
@svandenb-dev when I import edb into aedt, I don't see the new port. Can you double check it?

**Comment by ring630** [comment]
@svandenb-dev for 3dlayout application, do you have to define ref pingroup?

File: pyaedt/edb_core/components.py [hunk 1]
```diff
@@ -730,6 +730,86 @@ def create_source_on_component(self, sources=None):
...
+        >>> edb = Edb(path_to_edb_file)
+        >>> pin = "AJ6"
+        >>> ref_pins = ["AM7", "AM4"]
+        Or to take all reference pins
+        >>> ref_pins = [pin for pin in list(edb.components["U2A5"].pins.values()) if pin.net_name == "GND"]
+        >>> edb.components.create_port_on_pins(refdes="U2A5", pins=pin, reference_pins=ref_pins)
+        >>> edb.save_edb()
+        >>> edb.close_edb()
+        """
+        if pins and reference_pins:
```
Comment (maxcapodi78) [inline pyaedt/edb_core/components.py:769]: if pins and reference_pins are mandatory remove the if here and remove the default value in the method definition

File: pyaedt/edb_core/components.py [hunk 2]
```diff
@@ -730,6 +730,86 @@ def create_source_on_component(self, sources=None):
...
+            if isinstance(pins, str):
+                pins = [pins]
+            if isinstance(reference_pins, str):
+                reference_pins = [reference_pins]
+            if isinstance(refdes, str):
```
Comment (maxcapodi78) [inline pyaedt/edb_core/components.py:774]: we should consider also the option of EDBComponent is passed (component object in pyaedt)

File: pyaedt/edb_core/components.py [hunk 3]
```diff
@@ -730,6 +730,86 @@ def create_source_on_component(self, sources=None):
...
+        >>> ref_pins = ["AM7", "AM4"]
+        Or to take all reference pins
+        >>> ref_pins = [pin for pin in list(edb.components["U2A5"].pins.values()) if pin.net_name == "GND"]
+        >>> edb.components.create_port_on_pins(refdes="U2A5", pins=pin, reference_pins=ref_pins)
+        >>> edb.save_edb()
+        >>> edb.close_edb()
+        """
+        if pins and reference_pins:
+            if isinstance(pins, str):
+                pins = [pins]
```
Comment (maxcapodi78) [inline pyaedt/edb_core/components.py:771]: we should consider also EDBPadstackInstance object is passed

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]

File: pyaedt/edb_core/components.py (hunk 2 above)
Comment (svandenb-dev) [reply pyaedt/edb_core/components.py]: done

File: pyaedt/edb_core/components.py (hunk 1 above)
Comment (svandenb-dev) [reply pyaedt/edb_core/components.py]: done

**Review by maxcapodi78 — APPROVED** [review]
LGTM

File: pyaedt/edb_core/components.py (hunk 3 above)
Comment (svandenb-dev) [reply pyaedt/edb_core/components.py]: done

File: pyaedt/edb_core/components.py [hunk 4]
```diff
@@ -730,6 +730,86 @@ def create_source_on_component(self, sources=None):
...
+                for ref_pin_name in reference_pins:
+                    cmp_ref_pin = [pin for pin in list(refdes.pins.values()) if ref_pin_name in pin.name]
+                    if cmp_ref_pin:
+                        ref_cmp_pins.append(cmp_ref_pin[0])
+                if not ref_cmp_pins:
+                    return
+                reference_pins = ref_cmp_pins
+            if not len([pin for pin in reference_pins if isinstance(pin, EDBPadstackInstance)]) == len(reference_pins):
+                return
+            group_name = f"group_{pins[0].net_name}_{pins[0].name}"
```
[the remark this answers is not in the document]
Comment (svandenb-dev) [reply pyaedt/edb_core/components.py]: done

**Pull request merged**